src/measure/measure.c

```diff
@@ -104,6 +104,7 @@ static int help(int argc, char *argv[], void *userdata) {
                "     --linux=PATH        Path to Linux kernel image file            %7$s .linux\n"
                "     --osrel=PATH        Path to os-release file                    %7$s .osrel\n"
                "     --cmdline=PATH      Path to file with kernel command line      %7$s .cmdline\n"
+               "     --firmware=PATH     Path to firmware image file                %7$s .efifw\n"
```

Comment:
hmm, i don't think we should measure the firmware at all btw. we are not going to pass it to the kernel after all, we are going to apply it and reboot, and it will hence already be applied when we invoke the kernel.

Reply:
yes but post boot, the remote attestation will fail if the firmware measurement does not match the expected. I think the firmware needs to be measured but I am not sure if that happens indirectly as a part of measuring the entire encrypted private guest memory. @haraldh @kraxel @agraf can you throw some lights?

Reply:
The firmware is measured at boot (including re-boot with the new firmware image).  It is part of the launch measurement of the confidential vm.  In non-CC context the firmware will measure itself and update PCR 0.

Reply:
So no need to measure this as a part of the UKI right as Lennart says?

Reply:
I think in CC case, entire guest private memory is measured as a part of the launch digest (since it defines the initial state). This also includes the fw.

Reply:
> So no need to measure this as a part of the UKI right as Lennart says?

Yes

Reply:
ok https://trustedfirmware-a.readthedocs.io/en/v2.11/design_documents/measured_boot.html makes it a little clear. firmware gets measured automatically.